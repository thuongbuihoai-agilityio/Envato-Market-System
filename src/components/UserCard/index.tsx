import { memo } from 'react';
import isEqual from 'react-fast-compare';

// Components
import { Box, Flex, Heading, Text, IconButton, Avatar } from '@chakra-ui/react';

// Icons
import { BookMark, ShareNodes } from '@app/assets/icons';

// Images
import { IMAGES } from '@app/constants/images';

// Types
import { TEmployee } from '@app/interfaces';

// Utils
import { formatDecimalNumber } from '@app/utils';

interface userCardProps {
  user: TEmployee;
}

const UserCardComponent = ({ user }: userCardProps) => {
  const {
    firstName,
    lastName,
    position,
    lastActive,
    lastPlace,
    level,
    workTime,
    avatarURL,
    salary,
  } = user || {};

  const iconButtonStyles = {
    variant: 'outline',
    padding: 2,
    rounded: '50%',
    borderColor: 'secondary.350',
    w: 10,
    h: 10,
    _hover: {
      bg: 'primary.500',
      borderColor: 'primary.500',
      svg: { stroke: '#FFF' },
    },
  };

  return (
    <Box px={12} pb={7} bg="background.component.primary" rounded="lg">
      <Avatar
        src={avatarURL}
        borderRadius="xl"
        borderWidth="1px"
        w={16}
        h={16}
        pos="relative"
        left="calc(50% - 32px)"
        top={-8}
      />
      <Flex direction="column" alignItems="center" pb={7} mt={-4}>
        <Heading as="h3" variant="headingXl" color="text.octonary">
          {`${firstName} ${lastName}`}
        </Heading>
        <Text color="text.nonary">
          {`${position} • ${lastPlace} • ${lastActive}`}
        </Text>
        <Flex mt={6} gap={4}>
          <IconButton
            aria-label="Book Mark"
            icon={<BookMark />}
            {...iconButtonStyles}
          />
          <IconButton
            aria-label="Share Nodes"
            icon={<ShareNodes />}
            {...iconButtonStyles}
          />
        </Flex>
      </Flex>
      <Flex
        direction="column"
        gap={6}
        py={7}
        borderY="solid 1px"
        borderColor="border.senary"
      >
        <Flex alignItems="center" justifyContent="space-between">
          <Text variant="textSm" color="text.denary">
            Experience
          </Text>
          <Text variant="textSm" fontWeight="semibold" color="text.primary">
            2-4 Years
          </Text>
        </Flex>
        <Flex alignItems="center" justifyContent="space-between">
          <Text variant="textSm" color="text.denary">
            Seniority Level
          </Text>
          <Text variant="textSm" fontWeight="semibold" color="text.primary">
            {level}
          </Text>
        </Flex>
        <Flex alignItems="center" justifyContent="space-between">
          <Text variant="textSm" color="text.denary">
            Employment
          </Text>
          <Text variant="textSm" fontWeight="semibold" color="text.primary">
            {workTime}
          </Text>
        </Flex>
        <Flex alignItems="center" justifyContent="space-between">
          <Text variant="textSm" color="text.denary">
            Salary
          </Text>
          <Text variant="textSm" fontWeight="semibold" color="text.primary">
            ${formatDecimalNumber(salary, true)}
          </Text>
        </Flex>
      </Flex>
      <Box pt={6}>
        <Text variant="textSm" color="text.denary">
          Hiring Agent
        </Text>
        <Flex gap={4} mt={4}>
          <Avatar src={IMAGES.AVATAR.url} w={10} h={10} />
          <Box>
            <Text variant="textSm" fontWeight="semibold" color="text.primary">
              Annette Black
            </Text>
            <Text color="primary.500" variant="textSm">
              HR Specialist •{' '}
              <Text as="span" color="text.nonary" variant="textSm">
                from last week
              </Text>
            </Text>
          </Box>
        </Flex>
      </Box>
    </Box>
  );
};

const UserCard = memo(UserCardComponent, isEqual);

export default UserCard;
