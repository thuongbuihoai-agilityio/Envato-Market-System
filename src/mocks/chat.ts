// Constants
import { AVATAR_POSITION, IMAGES } from '@app/constants';
import { MESSAGE_TIME } from '.';

// TODO: Update later
export const USER_CHATS = [
  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hi, I need more information',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 1000,
  },

  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hello, I want to know more about your product',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 3000,
  },

  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hello, I want to know more about your product',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 6000,
  },

  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Hi, i am here for help',
    avatarPosition: AVATAR_POSITION.AFTER,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 2000,
  },

  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Sure, I can help you with that',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 4000,
  },
  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Sure, I can help you with that',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 7000,
  },
];

export const MEMBER_CHATS = [
  { uid: 'admin', name: 'Admin' },
  { uid: 'user1', name: 'User 1' },
];
