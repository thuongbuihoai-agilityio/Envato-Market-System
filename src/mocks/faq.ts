export const FAQ_DATA = [
  {
    id: 1,
    question: 'What is the most important part of a dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 2,
    question: 'What are the three types of dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 3,
    question: 'What are examples of admin?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 4,
    question: 'What are 5 benefits of dashboards?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 5,
    question: 'What makes a good dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 6,
    question: 'What are the 3 basic skills of administrator?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
];
