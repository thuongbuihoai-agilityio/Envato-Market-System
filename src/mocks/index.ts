export * from './spending';
export * from './select';
export * from './table';
export * from './sidebar';
