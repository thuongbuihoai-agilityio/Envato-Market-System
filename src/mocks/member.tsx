import { Image } from '@chakra-ui/react';

// Constants
import { IMAGES } from '@app/constants';

// Utils
import { MESSAGE_TIME_FORMAT } from '.';

export const MEMBER_CHAT = [
  {
    id: 'user1',
    avatar: 'https://example.com/avatar1.jpg',
    name: 'Kien Nguyen',
    status: 'Online',
    localeTime: MESSAGE_TIME_FORMAT,
    icon: (
      <Image
        w={5}
        h={5}
        src="icons/file.svg"
        alt={IMAGES.ATTACH.alt}
        fallbackSrc={IMAGES.FALLBACK.url}
      />
    ),
    statusColor: 'online',
  },
  {
    id: 'user2',
    avatar: 'https://example.com/avatar2.jpg',
    name: 'Huy Pham',
    status: 'Away',
    localeTime: MESSAGE_TIME_FORMAT,

    icon: (
      <Image
        w={5}
        h={5}
        src="icons/file.svg"
        alt={IMAGES.ATTACH.alt}
        fallbackSrc={IMAGES.FALLBACK.url}
      />
    ),
    statusColor: 'offline',
  },
];
