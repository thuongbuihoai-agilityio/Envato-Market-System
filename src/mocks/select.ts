export const SOCIAL_PLATFORM_OPTIONS = [
  {
    value: 'github',
    label: 'github',
  },
  {
    value: 'facebook',
    label: 'facebook',
  },
  {
    value: 'instagram',
    label: 'instagram',
  },
];
