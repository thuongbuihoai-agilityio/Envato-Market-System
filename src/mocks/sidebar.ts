// constants
import {
  HELP_ITEM_LIST,
  MENU_ITEM_LIST,
  OTHER_ITEM_LIST,
} from '@app/constants/sidebar';

export const EXPAND_SIDEBAR_MENU_LIST = [
  {
    id: 1,
    title: 'Menu',
    listItem: MENU_ITEM_LIST,
  },
  {
    id: 2,
    title: 'Help',
    listItem: HELP_ITEM_LIST,
  },
  {
    id: 3,
    title: 'Others',
    listItem: OTHER_ITEM_LIST,
  },
];
