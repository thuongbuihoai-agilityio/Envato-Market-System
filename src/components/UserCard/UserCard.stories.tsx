import { StoryObj, Meta } from '@storybook/react';

// Components
import UserCard from '.';

// Mocks
import { INITIAL_USER } from '@app/mocks';

const meta: Meta<typeof UserCard> = {
  title: 'Custom Components/UserCard',
  tags: ['autodocs'],
  component: UserCard,
  argTypes: {
    user: {
      description: 'Information of user',
    },
  },
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;
type Story = StoryObj<typeof UserCard>;

export const Default: Story = {
  args: {
    user: INITIAL_USER,
  },
};
