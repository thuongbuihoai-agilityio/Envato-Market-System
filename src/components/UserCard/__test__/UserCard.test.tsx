import { render } from '@testing-library/react';

// Components
import UserCard from '..';

// Mock
import { INITIAL_USER } from '@app/mocks';

describe('UserCard component', () => {
  it('renders correctly', () => {
    const { container } = render(<UserCard user={INITIAL_USER} />);

    expect(container).toMatchSnapshot();
  });
});
