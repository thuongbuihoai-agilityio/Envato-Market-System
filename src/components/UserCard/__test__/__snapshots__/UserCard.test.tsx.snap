// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component renders correctly 1`] = `
<div>
  <div
    class="css-13dage6"
  >
    <span
      class="chakra-avatar css-1otlzun"
    >
      <svg
        aria-label=" avatar"
        class="chakra-avatar__svg css-16ite8i"
        role="img"
        viewBox="0 0 128 128"
      >
        <path
          d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
          fill="currentColor"
        />
        <path
          d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="css-1sdqwkj"
    >
      <h3
        class="chakra-heading css-1u7vw38"
      >
         
      </h3>
      <p
        class="chakra-text css-1uvw9jh"
      >
         •  • 
      </p>
      <div
        class="css-1mcd17k"
      >
        <button
          aria-label="Book Mark"
          class="chakra-button css-h640nb"
          type="button"
        >
          <svg
            fill="none"
            height="25"
            stroke="#a0aec0"
            viewBox="0 0 24 25"
            width="24"
          >
            <path
              d="M19 21.5659L12 16.5659L5 21.5659V5.56592C5 5.03548 5.21071 4.52678 5.58579 4.1517C5.96086 3.77663 6.46957 3.56592 7 3.56592H17C17.5304 3.56592 18.0391 3.77663 18.4142 4.1517C18.7893 4.52678 19 5.03548 19 5.56592V21.5659Z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </button>
        <button
          aria-label="Share Nodes"
          class="chakra-button css-h640nb"
          type="button"
        >
          <svg
            fill="none"
            height="21"
            stroke="#a0aec0"
            viewBox="0 0 20 21"
            width="20"
          >
            <path
              d="M4 13.5659C5.65685 13.5659 7 12.2228 7 10.5659C7 8.90906 5.65685 7.56592 4 7.56592C2.34315 7.56592 1 8.90906 1 10.5659C1 12.2228 2.34315 13.5659 4 13.5659Z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M16 7.56592C17.6569 7.56592 19 6.22277 19 4.56592C19 2.90906 17.6569 1.56592 16 1.56592C14.3431 1.56592 13 2.90906 13 4.56592C13 6.22277 14.3431 7.56592 16 7.56592Z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M16 19.5659C17.6569 19.5659 19 18.2228 19 16.5659C19 14.9091 17.6569 13.5659 16 13.5659C14.3431 13.5659 13 14.9091 13 16.5659C13 18.2228 14.3431 19.5659 16 19.5659Z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M6.69995 9.26572L13.3 5.86572"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M6.69995 11.8657L13.3 15.2657"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="css-iofhnq"
    >
      <div
        class="css-1lekzkb"
      >
        <p
          class="chakra-text css-80l8so"
        >
          Experience
        </p>
        <p
          class="chakra-text css-11c5ex2"
        >
          2-4 Years
        </p>
      </div>
      <div
        class="css-1lekzkb"
      >
        <p
          class="chakra-text css-80l8so"
        >
          Seniority Level
        </p>
        <p
          class="chakra-text css-11c5ex2"
        />
      </div>
      <div
        class="css-1lekzkb"
      >
        <p
          class="chakra-text css-80l8so"
        >
          Employment
        </p>
        <p
          class="chakra-text css-11c5ex2"
        />
      </div>
      <div
        class="css-1lekzkb"
      >
        <p
          class="chakra-text css-80l8so"
        >
          Salary
        </p>
        <p
          class="chakra-text css-11c5ex2"
        >
          $250-300
        </p>
      </div>
    </div>
    <div
      class="css-8zy5qf"
    >
      <p
        class="chakra-text css-80l8so"
      >
        Hiring Agent
      </p>
      <div
        class="css-2e52m3"
      >
        <span
          class="chakra-avatar css-6id39j"
        >
          <svg
            aria-label=" avatar"
            class="chakra-avatar__svg css-16ite8i"
            role="img"
            viewBox="0 0 128 128"
          >
            <path
              d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
              fill="currentColor"
            />
            <path
              d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-11c5ex2"
          >
            Annette Black
          </p>
          <p
            class="chakra-text css-ss2gmn"
          >
            HR Specialist •
             
            <span
              class="chakra-text css-1uvw9jh"
            >
              from last week
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
